src/metric/regression_metric.hpp

```diff
@@ -171,6 +171,19 @@ class L1Metric: public RegressionMetric<L1Metric> {
   }
 };
 
+/*! \brief SMAPE loss for regression task */
+class SMAPEMetric: public RegressionMetric<SMAPEMetric> {
+public:
+  explicit SMAPEMetric(const MetricConfig& config) :RegressionMetric<SMAPEMetric>(config) {}
+
+  inline static double LossOnPoint(label_t label, double score, const MetricConfig&) {
+    return std::fabs(score - label) / (std::fabs(label) + std::fabs(score));
```

Comment:
I think ```std::fabs(label) + std::fabs(score)``` also could be zero.

Reply:
Yes you are correct. I am working on implementing this again and will update the PR sometime this weekend.

Reply:
great! thanks very much